app/services/db/identity/total_identities_per_ial_count.rb

```diff
@@ -0,0 +1,12 @@
+module Db
+  module Identity
+    class TotalIdentitiesPerIalCount
+      def self.call(ial)
+        sql = <<~SQL
+          SELECT COUNT(*) FROM identities WHERE ial=#{ial}
+        SQL
+        ActiveRecord::Base.connection.execute(sql)[0]['count'].to_i
+      end
```

Comment:
this is just a single method, only used in one class, can we just add this as a method to the report?

Reply:
negative.  these are all reusable db routines.

Reply:
yes, functions/methods are great ways to reuse code. What I'm asking is it we can move them in to the report file, because that's the only place these are called from? There's no code outside the report that uses them, so packaging them as a class seems a little premature?

Reply:
Disagree on this one.  How would you know there is a reusable db routine if it was inlined somewhere?  The way it's structured now it's effectively a self documenting API for dealing with the db.

Reply:
I don't have strong feelings on it, but I am lazy so I prefer having things in less files when possible!

Reply:
The structure gives you a self documenting api.  If you need something from identities you go to db/identity and see if there is a routine there
![Screen Shot 2022-05-13 at 11 19 00 AM](https://user-images.githubusercontent.com/34244063/168325731-d563ac5f-e98f-45a7-b8b7-16d4b338831e.png)

Reply:
In an API driven world any or all of these routines can be automatically wrapped and exposed to the front end (hence why we are using basic types).  Moreover you are increasing team parallelism/less merges/conflicts.  Finally, if I'm going to write a complicated routine to get data from say identities how would I know that one does not exist if it's inlined somewhere?